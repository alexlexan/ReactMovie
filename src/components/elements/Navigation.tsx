import React from "react";
import { Link } from "react-router-dom";

import { StyledNavigation } from "../styles/StyledNavigation";

type Props = {
  movie: string;
};

const Navigation: React.FC<Props> = ({ movie }) => (
  <StyledNavigation>
    <div className="navigation-content">
      <Link to="/">
        <p>Home</p>
      </Link>
      <p>|</p>
      <p>{movie}</p>
    </div>
  </StyledNavigation>
);

export default Navigation;
