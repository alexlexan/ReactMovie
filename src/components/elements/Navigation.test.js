import React from "react";
import Navigation from "./Navigation";

const setUp = (props) => shallow(<Navigation {...props} />);

describe("should render Navigation component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render Navigation component", () => {
    const Navigation = component.find(".navigation-content");
    expect(Navigation.length).toBe(1);
  });

  it("should render Navigation component props", () => {
    const comp = shallow(<Navigation />);
    expect(comp).toMatchSnapshot();
  });

  it("should render Navigation component with default props", () => {
    const comp = shallow(<Navigation movie="Default value" />);
    expect(comp).toMatchSnapshot();
  });
});
