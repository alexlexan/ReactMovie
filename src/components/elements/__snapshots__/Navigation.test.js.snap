// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navigation component should render Navigation component props 1`] = `
<styled.div>
  <div
    className="navigation-content"
  >
    <Link
      to="/"
    >
      <p>
        Home
      </p>
    </Link>
    <p>
      |
    </p>
    <p />
  </div>
</styled.div>
`;

exports[`should render Navigation component should render Navigation component with default props 1`] = `
<styled.div>
  <div
    className="navigation-content"
  >
    <Link
      to="/"
    >
      <p>
        Home
      </p>
    </Link>
    <p>
      |
    </p>
    <p>
      Default value
    </p>
  </div>
</styled.div>
`;
